File: exercism/go/archive/raindrops/raindrops.go
// Package raindrops is a factoring function that tells us if we're divisible by 3, 5, 7 inclusive or none at all!
package raindrops

import (
	"bytes"
	"strconv"
)

// testVersion confirms the version we're testing against
const testVersion = 2

// Convert is our function to handle the input (int), and convert to a string based on whether its divisible
// by 3, 5, 7 inclusive (or none at all)
func Convert(theval int) string {
	var output bytes.Buffer

	if (theval % 3) == 0 {
		output.WriteString("Pling")
	}

	if (theval % 5) == 0 {
		output.WriteString("Plang")
	}

	if (theval % 7) == 0 {
		output.WriteString("Plong")
	}

	if output.Len() == 0 {
		return strconv.Itoa(theval)
	}

	return output.String()
}

File: exercism/python/raindrops/raindrops.py
"""Module to calculate the Raindrops module tests."""


def convert(number: int) -> str:
    """Calculate the Raindrops value for the provided number.

    :param number: int - number to convert from value to string, if divisible by any of 3, 5, or 7
    :return: str - return a string representing Raindrops if divisible by 3, 5, or 7; otherwise initial value
    """

    output = ""
    if number % 3 == 0:
        output += "Pling"

    if number % 5 == 0:
        output += "Plang"

    if number % 7 == 0:
        output += "Plong"

    if not output:
        return f"{number}"

    return output
